shared/util.py

```diff
@@ -1195,6 +1195,158 @@ def update_organization_subscription(
                 f"[util__module] update_organization_subscription: {organization_id} not found"
             )
 
+def update_subscription_logs(
+    subscription_id,
+    action,
+    previous_plan=None,
+    current_plan=None,
+    modified_by=None,
+    modified_by_name=None,
+    status_financial_assistant=None
+):
+    """
+        Logs subscription-related events to the audit container.
+
+        Parameters:
+        - subscription_id (str): Subscription ID.
+        - action: New Subscription, Subscription Tier Change, Financial Assistant On, Financial Assistant Off.
+        - previous_plan: Pre-plan before a plan change.
+        - current_plan: Current plan after a plan change.
+        - modified_by: UserID who made the modification
+        - modified_by_user: UserName who made the modification
+    """
+    if not subscription_id:
+        return {"error": "Subscription ID not provided."}
+
+    logging.info(f"Logging audit for subscription: {subscription_id}")
+
+    credential = DefaultAzureCredential()
+    db_client = CosmosClient(AZURE_DB_URI, credential, consistency_level="Session")
+    db = db_client.get_database_client(database=AZURE_DB_NAME)
+
+    #Containers
+    organizations_container = db.get_container_client("organizations")
+    auditlogs_container = db.get_container_client("auditLogs")
+
+    try:
+        #Query the organization using the subscription_id
+        query = "SELECT * FROM c WHERE c.subscriptionId = @subscription_id"
+        parameters = [{"name": "@subscription_id", "value": subscription_id}]
+        result = list(
+            organizations_container.query_items(
+                query=query, parameters=parameters, enable_cross_partition_query=True
+            )
+        )
+
+        if result:
+            organization = result[0]
+            organization_name = organization.get("name")
+            organization_owner = organization.get("owner")
+
+            if not organization_name or not organization_owner:
+                logging.warning(f"Missing required fields in organization: {organization}")
+                return {"error": "Missing 'name' or 'owner' fields in organization data."}
+
+            audit_log_entry = {
+                "id": str(uuid.uuid4()),  # Unique ID for the audit log
+                "organizationName": organization_name,
+                "organizationOwner": organization_owner,
+                "subscriptionId": subscription_id,
+                "action": action,
+                "changeTime": int(datetime.utcnow().timestamp()),
+            }
+
+            # Add previous and current plans if the action is "Subscription Tier Change"
+            if action == "Subscription Tier Change":
```

Comment:
@ramirezmorac2 you might wan t to define constants for actions to avoid hardcoding and typos:

```python
class SubscriptionActions:
    SUBSCRIPTION_TIER_CHANGE = "Subscription Tier Change"
    NEW_SUBSCRIPTION = "New Subscription"
    FINANCIAL_ASSISTANT_CHANGE = "Financial Assistant Change"
    ```